extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/ItagItem.java

```diff
@@ -159,22 +173,58 @@ public ItagItem(final int id,
         this.avgBitrate = avgBitrate;
     }
 
-    private final MediaFormat mediaFormat;
-
+    /**
+     * Copy constructor of the {@link ItagItem} class.
+     *
+     * @param itagItem the {@link ItagItem} to copy its properties into a new {@link ItagItem}
+     */
+    public ItagItem(@Nonnull final ItagItem itagItem) {
+        this.mediaFormat = itagItem.mediaFormat;
+        this.id = itagItem.id;
+        this.itagType = itagItem.itagType;
+        this.avgBitrate = itagItem.avgBitrate;
+        this.sampleRate = itagItem.sampleRate;
+        this.audioChannels = itagItem.audioChannels;
+        this.resolutionString = itagItem.resolutionString;
+        this.fps = itagItem.fps;
+        this.bitrate = itagItem.bitrate;
+        this.width = itagItem.width;
+        this.height = itagItem.height;
+        this.initStart = itagItem.initStart;
+        this.initEnd = itagItem.initEnd;
+        this.indexStart = itagItem.indexStart;
+        this.indexEnd = itagItem.indexEnd;
+        this.quality = itagItem.quality;
+        this.codec = itagItem.codec;
+        this.targetDurationSec = itagItem.targetDurationSec;
+        this.approxDurationMs = itagItem.approxDurationMs;
+        this.contentLength = itagItem.contentLength;
+    }
 
     public MediaFormat getMediaFormat() {
         return mediaFormat;
     }
 
+    private final MediaFormat mediaFormat;
+
     public final int id;
     public final ItagType itagType;
 
     // Audio fields
-    public int avgBitrate = -1;
+    /** @deprecated Use {@link #getAverageBitrate()} instead. */
+    @Deprecated
```

Comment:
I am still not sure whether we really want to leave this... I would just remove them and call it a day. Clients need to switch to getters and that's it. **Comment for after PR is merged!**